R/estimatr_lh_robust.R

```diff
@@ -0,0 +1,161 @@
+#' variables with large numbers of groups and when using "HC1" or "stata" standard errors.
+#' See 'Details'.
+#' @param se_type The sort of standard error sought. If \code{clusters} is
+#' not specified the options are "HC0", "HC1" (or "stata", the equivalent),
+#'  "HC2" (default), "HC3", or
+#' "classical". If \code{clusters} is specified the options are "CR0", "CR2" (default), or "stata". Can also specify "none", which may speed up estimation of the coefficients.
+#' @param ci logical. Whether to compute and return p-values and confidence
+#' intervals, TRUE by default.
+#' @param alpha The significance level, 0.05 by default.
+#' @param return_vcov logical. Whether to return the variance-covariance
+#' matrix for later usage, TRUE by default.
+#' @param try_cholesky logical. Whether to try using a Cholesky
+#' decomposition to solve least squares instead of a QR decomposition,
+#' FALSE by default. Using a Cholesky decomposition may result in speed gains, but should only
+#' be used if users are sure their model is full-rank (i.e., there is no
+#' perfect multi-collinearity)
+#' @param linearHypothesis A character string or a matrix specifying combination.
+#' See \code{\link{linearHypothesis}} for more details.
+#' @param lh_term A character string indicating the name of term to be assigned to the linear combination.
+#' @details
+#'
+#' This function is a wrapper for \code{\link{lm_robust}} and for
+#' \code{\link{car::linearHypothesis}}. It first runs \code{lm_robust} and
+#' then passes \code{"lm_robust"} object as an argument to \code{car::linearHypothesis}.
+#'
+#' @return An object of class \code{"lm_robust"} and \code{"lh"}.
+#'
+#'
+#' Both objects are a list containing the
+#' following components:
+#'   \item{coefficients}{the estimated linear combination of coefficients}
+#'   \item{std.error}{the estimated standard errors of the linear combination}
+#'   \item{statistic}{the t-statistic}
+#'   \item{df}{the estimated degrees of freedom}
+#'   \item{p.value}{the p-values from a two-sided t-test using \code{coefficients}, \code{std.error}, and \code{df}}
+#'   \item{conf.low}{the lower bound of the \code{1 - alpha} percent confidence interval}
+#'   \item{conf.high}{the upper bound of the \code{1 - alpha} percent confidence interval}
+#'   \item{term}{a character vector of coefficient names}
+#'   \item{alpha}{the significance level specified by the user}
+#' @export
+#'
+lh_robust <- function(formula,
+                      data,
+                      weights,
+                      subset,
+                      clusters,
+                      fixed_effects,
+                      se_type = NULL,
+                      ci = TRUE,
+                      alpha = .05,
+                      return_vcov = TRUE,
+                      try_cholesky = FALSE,
+                      linearHypothesis = NULL,
+                      lh_term = NULL
+                      ) {
+
+  return_list <- list()
+
+
+  datargs <- enquos(
```

Comment:
instead of copy-pasting all of lm_robust here, just call it.

Reply:
couldn't avoid this.

Reply:
Why can't we just call lm_robust and then have lh_robust add to the output?